### ISSUE TITLE
Additional vanishers?

### ISSUE THREAD

**Issue by laryn** [backdrop-contrib/gdpr_cookies#6]
I'm curious how to add additional vanishers/services to this list?

![Screen Shot 2019-04-23 at 10 35 31 AM](https://user-images.githubusercontent.com/871421/56595089-a0a17d00-65b3-11e9-88f6-c56c1c6d5bdf.jpg)

For example, if an organization uses Recaptcha, MapBox, or another third party service that isn't listed in the dropdown, how do we use this?

**Comment by Graham-72** [comment]
@laryn I believe the third party services are provided in the tarteaucitron script
https://github.com/backdrop-contrib/gdpr_cookies/blob/master/js/tarteaucitron/tarteaucitron.services.js

It looks as though the current version provides for Recaptcha at https://github.com/AmauriC/tarteaucitron.js/blob/master/tarteaucitron.services.js

Hopefully we can update gdpr_cookies soon - issue #5.

**Comment by Graham-72** [comment]
@laryn I have been trying to understand how this is done, as inherited from blizz_vanisher (the parent of this module). It seems that the free (open source) version of the tarteaucitron script requires manual substitution of code for each service. Blizz_vanisher provides a set of php scripts to automate this, each third party service having dedicated 'vanisher' script within this module. Writing one of these looks difficult but there are several examples to follow when someone has the inclination and time.

**Comment by djzwerg** [comment]
It would be nice to have Instagram and OpenStreetMap added :-)

**Comment by djzwerg** [comment]
Is there any work on this issue?

**Comment by olafgrabienski** [comment]
I stumbled about this issue because I probably need the Instagram service of the library for an upcoming site. If I don't miss anything, Instagram is part of the [current library version](https://github.com/AmauriC/tarteaucitron.js/releases/tag/v1.12.0) (not sure when it was added). @laryn, after #19, are you maybe up to update the library once more?

However, even with an up to date library, we need to figure out how to make a service available in the Backdrop module settings. As @Graham-72 pointed out in https://github.com/backdrop-contrib/gdpr_cookies/issues/6#issuecomment-508249585, there is some extra work to do. As usual, I'm not able to do the coding, but I'd be happy to discuss approaches and test a PR.

**Comment by nattywebdev** [comment]
I'm hoping that someone has figured out how to add extra vanishers to make this module of some use! I need to add one for an existing website where (for example) external service from Elfsight.com are used, with corresponding cookies. It appears that the elfsight cookies are included in https://apps.elfsight.com/p/platform.js so I hope it's just that I need a way to connect that to the module.
Any clues would be most welcome!

**Comment by yorkshire-pudding** [comment]
@nattywebdev - I have a reasonable understanding of how to create additional vanishers but I need to focus on paid work and things that support my paid work.
Knowing your ability level, I'm not sure you would be able to add a vanisher yourself, but the general approach is to extend one of the abstract types of vanisher to a specific one.  Perhaps, see if there is a Drupal integration and see which cookie service it uses; if it is Blizz Vanisher then that might make it easier.

**Comment by nattywebdev** [comment]
@yorkshire-pudding thanks for that. I would be up for working on a vanisher for that example, based on an existing one if a suitable example exists. Maybe useful if there was documentation advising how to create a new vanisher so that this module could be extended for wider use. Doesn't seem to be any existing Drupal instances using the 'elfsight' cookies. Elfsight provides a functional Instagram feed that can be easily added to any site, as well as many other such feeds.

**Comment by yorkshire-pudding** [comment]
> Elfsight provides a functional Instagram feed that can be easily added to any site, as well as many other such feeds.

So not an optician service for the elven community then 😉
